File: Mobile Client/lib/SiteEngineer/Comment.dart
import 'dart:async';
import 'dart:convert';

import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;



class Comment {
  final int commentId;
  final int taskId;
  final String taskName;
  final int userId;
  final String commentText;
  final String commentDate;

  Comment({
    required this.commentId,
    required this.taskId,
    required this.taskName,
    required this.userId,
    required this.commentText,
    required this.commentDate,
  });
  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      commentId: json['commentId'],
      taskId: json['taskId'],
      taskName: json['taskName'] ??'',
      userId: json['userId'],
      commentText: json['commentText']??'',
      commentDate: json['commentDate']??'',
    );
  }

    Map<String, dynamic> toJson() {
    return {
      'commentId': commentId,
      'taskId': taskId,
      'taskName': taskName,
      'userId': userId,
      'commentText': commentText,
      'commentDate': commentDate,
    };
  }

  
}

File: Server/src/main/java/com/Ascendia/server/entity/SiteManager/Comment.java
package com.Ascendia.server.entity.SiteManager;

import com.Ascendia.server.entity.Administrator.User;
import com.Ascendia.server.entity.ProjectManager.Task;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "task_comment_table")
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "commentId")
    private int commentId;

    @ManyToOne
    @JoinColumn(name = "task_id", nullable = false, referencedColumnName = "taskId")
    private Task task;

    //@Column(name = "taskName")
    //private String taskName;

    @ManyToOne
    @JoinColumn(name = "commentor_user_id", nullable = false, referencedColumnName = "userID")
    private User commentedUser;

    @Column(name = "commentText", length = 500)
    private String commentText;

    //@Column(name = "commentDate")
    //private String commentDate;

    @Column(name = "commentDateTime")
    private LocalDateTime commentDate;

}